reference/performance.mdx

```diff
@@ -5,36 +5,288 @@ sidebarTitle: Performance
+| [Mixed data types in the same field](#mixing-unrelated-data-in-datasets) | Moderate | Reduces compression, complicates queries |
+| [Using regex when simpler filters suffice](#regular-expressions-when-simple-filters-suffice) | Moderate | More CPU-heavy scanning |
+| [Overusing runtime JSON parsing (parse_json)](#overusing-runtime-json-parsing-parse_json) | Moderate | CPU overhead, no indexing on nested fields |
+| [Virtual fields for simple transformations](#virtual-fields-for-simple-transformations) | Low | Extra overhead for trivial conversions |
+| [Poor filter order in queries](#poor-filter-order-in-queries) | Low | Suboptimal scanning of data |
 
-## Optimize datasets
+## Mixing unrelated data in datasets
+
+### Problem
+
+A “kitchen-sink” dataset is one in which events from multiple, unrelated applications or services get lumped together, often resulting in:
+
+- **Excessive width (too many columns)**: Adding more and more unique fields bloats the schema, reducing query throughput.
+- **Mixed data types in the same field**: For example, some events store `user_id` as a string, while others store it as a number in the same `user_id` field.
+- **Unrelated schemas in a single dataset**: Columns that make sense for one app might be `null` or typed differently for another.
+
+These issues reduce compression efficiency, make indexing less effective, and force Axiom to scan more data than necessary.
+
+### Why it matters
+
+- **Slower queries**: Each query must parse wider blocks of data and handle inconsistent column types.
+- **Higher resource usage**: Wide schemas reduce row packing in blocks, harming throughput and potentially raising costs.
+- **Harder data exploration**: When fields differ drastically between events, discovering the correct columns or shaping queries becomes more difficult.
+
+### How to fix it
+
+- **Keep datasets narrowly focused:** Group data from the same application or service in its own dataset. For example, keep `k8s_logs` separate from `web_traffic`.
+- **Avoid mixing data types for the same field:** Enforce consistent types during ingest. If a field is numeric, always send numeric values.
+- **Consider using map fields:** If you have sparse or high-cardinality nested data, consider storing it in a single map (object) field instead of flattening every key. This reduces the total number of top-level fields. Axiom’s [map fields](/apl/data-types/map-fields#map-fields) are optimized for large objects.
+
+## Excessive backfilling and large `_time` vs. `_sysTime` gaps
+
+### Problem
+
+Axiom’s index is primarily built around time. Ideally, incoming events for a block lie in a closely bounded time range. However, backfilling large amounts of historical data after the fact (especially out of chronological order) creates wide time overlaps in blocks. If `_time` is far from `_sysTime` (the time the event was ingested), Axiom’s primary time index is weakened.
```

Comment:
Removing ambiguity. We don't have other indexes at the moment. So, "primary" doesn't make sense.

```suggestion
Axiom’s `_time` index is critical for query performance. Ideally, incoming events for a block lie in a closely bounded time range. However, backfilling large amounts of historical data after the fact (especially out of chronological order) creates wide time overlaps in blocks. If `_time` is far from `_sysTime` (the time the event was ingested), Axiom’s time index effectiveness is weakened.
```